front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/RedBlackToBlueGreenK8sPipelinesMigration.java

```diff
@@ -0,0 +1,115 @@
+
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2023, 6, 1);
+
+  private final PipelineDAO pipelineDAO;
+
+  public RedBlackToBlueGreenK8sPipelinesMigration(PipelineDAO pipelineDAO) {
+    this.pipelineDAO = pipelineDAO;
+  }
+
+  @Override
+  public boolean isValid() {
+    return LocalDate.now(Clock.systemDefaultZone()).isBefore(VALID_UNTIL);
+  }
+
+  @Override
+  public void run() {
+    log.info(
+        "Starting the migration of K8s pipelines using RED/BLACK traffic management strategy to BLUE/GREEN");
+    pipelineDAO.all().stream()
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration.pipelineWithRedBlackStrategyPredicate())
+        .forEach(this::migrate);
+  }
+
+  private void migrate(Pipeline pipeline) {
+    List<Map<String, Object>> stages = new ArrayList<>(pipeline.getStages());
+
+    stages.stream()
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::kubernetesProvider)
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::deployManifestStage)
+        .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagement)
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::trafficManagementEnabled)
+        .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagementOptions)
+        .forEach(
+            trafficManagementOptions -> {
+              Object strategy = trafficManagementOptions.get("strategy");
+              if ("redblack".equals(strategy)) {
+                trafficManagementOptions.put("strategy", "bluegreen");
+              }
+            });
+
+    this.pipelineDAO.update(pipeline.getId(), pipeline);
+    log.info(
+        "Migrated pipeline (application: {}, pipelineId: {}, name: {})",
+        pipeline.getApplication(),
+        pipeline.getId(),
+        pipeline.getName());
+  }
+
+  private static Predicate<Pipeline> pipelineWithRedBlackStrategyPredicate() {
+    return pipeline ->
+        pipeline.getStages().stream()
+            .filter(RedBlackToBlueGreenK8sPipelinesMigration::kubernetesProvider)
+            .filter(RedBlackToBlueGreenK8sPipelinesMigration::deployManifestStage)
+            .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagement)
+            .filter(RedBlackToBlueGreenK8sPipelinesMigration::trafficManagementEnabled)
+            .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagementOptions)
+            .anyMatch(RedBlackToBlueGreenK8sPipelinesMigration::redBlackStrategy);
+  }
+
+  private static Map<String, Object> getTrafficManagementOptions(
+      Map<String, Object> trafficManagement) {
+    return (Map<String, Object>) trafficManagement.get("options");
+  }
+
+  private static boolean trafficManagementEnabled(Map<String, Object> trafficManagement) {
+    return Boolean.TRUE.equals(trafficManagement.get("enabled"));
+  }
+
+  private static Map<String, Object> getTrafficManagement(Map<String, Object> stage) {
+    return (Map<String, Object>) stage.get("trafficManagement");
```

Comment:
Don't know if trafficManagement can be null, or if that'd break this (I THOUGHT streams filter handled that situation though), but might be worth a test of "no traffic management is set at all